### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by maasencioh**
Read csv files and return structure compatible with `common-spectra`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,10 +14,28 @@ File parser for n dimesional values.
...
+meta.second,,2
+temp, volt, curr
+1,1,1
+2,2,2
+3,3,3
+`;
+
+expect(ndParse(simpleExample, { separator: ',' })).toStrictEqual({
+  data: {
+    temp: { data: [1, 2, 3], label: 'temp' },
```
Comment (lpatiny) [inline README.md]: The 'symbol' should really be only one letter otherwise it breaks jcamp format

File: src/__tests__/test.ts [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+
+test('simpleExample', () => {
+  expect(ndParse(simpleExample)).toStrictEqual({
+    data: {
+      temp: { data: [1, 2, 3], label: 'temp' },
+      volt: { data: [1, 2, 3], label: 'volt' },
+      curr: { data: [1, 2, 3], label: 'curr' },
+    },
+    meta: {
+      'meta.first': '1',
```
Comment (lpatiny) [inline src/__tests__/test.ts:23]: May have to think about an 'autoTyping' option

File: src/index.ts [hunk 3]
```diff
@@ -1,7 +1,58 @@
...
+
+export function ndParse(text: string, options?: OptionsType): OutputType {
+  const { separator = ',' } = options || {};
+  let meta: OutputType['meta'] = {};
+  let data: OutputType['data'] = {};
+
+  let tempHeader: string[] | undefined;
+  let header: string[] | undefined;
+  for (const line of text.split(/\r\n|\r|\n/)) {
+    const lineItems = line.split(separator);
```
Comment (lpatiny) [inline src/index.ts]: I like the name `fields` for lineItems

**Comment by lpatiny** [comment]
You may merge the code after making the small changes. Also please make a `release-as: v0.0.1` (you know this commit message)

File: src/__tests__/test.ts [hunk 4]
```diff
@@ -1,7 +1,24 @@
...
-  it('should return 42', () => {
-    expect(myModule()).toBe(42);
+const simpleExample = `
+meta.first,1
+meta.second,,2
+temp, volt, curr
+1,1,1
+2,2,2
+3,3,3
+`;
```
Comment (targos) [inline src/__tests__/test.ts:13]: do you also have a "real" file that you could add as a public test case?

File: src/__tests__/test.ts (hunk 2 above)
Comment (maasencioh) [reply src/__tests__/test.ts:23]: I think that this kind of decisions should be taken in the library that will use it

**Pull request merged**